clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -187,42 +169,82 @@ public HubConnectionState getConnectionState() {
      *
      * @throws Exception An error occurred while connecting.
      */
-    public CompletableFuture start() throws Exception {
+    public CompletableFuture<Void> start() throws Exception {
```

Comment:
Why?

Reply:
"CompletableFuture is a raw type. References to generic type CompletableFuture<T> should be parameterized"
This is what VSCode tells me with a green squiggle, so changing to `<Void>` gets rid of it

Reply:
It's general cleanliness. I believe `CompletableFuture` is a shorthand for `CompletableFuture<?>` (remember Java doesn't allow multiple types with the same name but different generic parameters) which allows `start` to return **any** kind of `CompletableFuture`. This makes the type-checker ensure you return no value from your future.